### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Constructions.LimitsOfProductsAndEqualizers

### PR DESCRIPTION
**Pull request by mattrobball**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Constructions/LimitsOfProductsAndEqualizers.lean [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+-- See note [dsimp, simp].
+#align category_theory.limits.preserves_limit_of_preserves_equalizers_and_product CategoryTheory.Limits.preservesLimitOfPreservesEqualizersAndProduct
+
+end
+
+/- Porting note: the original parameter [∀ (J) [Fintype J], PreservesColimitsOfShape 
+(Discrete.{0} J) G] triggered the error "invalid parametric local instance, parameter 
+with type Fintype J does not have forward dependencies, type class resolution cannot 
+use this kind of local instance because it will not be able to infer a value for this 
+parameter." Changed to implicit below. -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Constructions/LimitsOfProductsAndEqualizers.lean:216]: Should the outer or the inner `[]` be changed to `{}`?
The same for the other instance of this problem lower in the file.

Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Constructions/LimitsOfProductsAndEqualizers.lean]: I tried that. I asked a question on Zulip: [invalid parametric local instance](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/invalid.20parametric.20local.20instance)

I also tried to refactor adding a new class `PreservesFiniteProducts` but that was too involved for the port (IMO). So I expanded the porting note. This (and the colimit version) only pop each once more for left exact functors. If that is a bear, I'll come back and fiddle with this more.

Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Constructions/LimitsOfProductsAndEqualizers.lean]: Ok. So I went back and did it as minimally as possible. I think this is the optimal solution.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**